src/main/scala/fs2/Chunk.scala

```diff
@@ -34,6 +34,16 @@ trait Chunk[+A] { self =>
     iterator.map(f).copyToBuffer(buf)
     Chunk.indexedSeq(buf)
   }
+  def mapAccumulate[S,B](s0: S)(f: (S,A) => (S,B)): (S,Chunk[(S,B)]) = {
```

Comment:
Return type should just be `(S, Chunk[B])`. If you want your `B` to incorporate the `S`, that is the job of the `f` function. (That is, `B` could be `(S,Blah)`.) So when you use this to implement `mapAccumulate` on streams, you'll have to do exactly that, since we want each value tagged with the current `S`.